### PR TITLE
Writeln

### PR DESCRIPTION
**Pull request by 12mohaned**
Simple refactor of Writeln proc, changed the return type to be bool instead of void. This pull request addresses issue [17179](https://github.com/chapel-lang/chapel/issues/17179).

### CODE REVIEW

**Review by mppf — APPROVED** [review]
This looks good to me but two things need to happen before merging it:

 * I need to run testing
 * You need to update the PR message to describe what the PR does. You could start by using the description in #17381. AFAIK there is also an issue that this resolves - #17179 - can you link that?

**Comment by mppf** [comment]
hi @12mohaned I've run testing and these files will need adjustment --

```
constrained-generics/basic/associated/using-acons-functions.chpl
constrained-generics/basic/set2/cgfun-call-cgfun-1.chpl
constrained-generics/basic/set2/ref-intents-1.chpl
constrained-generics/basic/set2/resolve-implements-types.chpl
constrained-generics/basic/set2/wrappers-for-impl-functions.chpl
constrained-generics/random/demo-random.chpl
constrained-generics/random/test-random.chpl
```

I suspect the fix is as simple as getting the interface declarations declaring `write` to have one that returns a `bool` but @vasslitvinov is the expert on these

File: modules/standard/ChapelIO.chpl [hunk 1]
```diff
@@ -834,12 +834,17 @@ module ChapelIO {
   }
 
   /* Equivalent to ``try! stdout.write``. See :proc:`IO.channel.write` */
-  proc write(const args ...?n) {
-    try! stdout.write((...args));
+  proc write(const args ...?n):bool{
+    try! {
+      return stdout.write((...args));
+    }
```
Comment (vasslitvinov) [inline modules/standard/ChapelIO.chpl:840]: I suggest we do not need these extra curly braces.
What is the benefit of adding them?

**Comment by vasslitvinov** [comment]
I believe Michael is correct. The change should be as simple as this:
```diff
--- a/test/constrained-generics/basic/set2/ref-intents-1.chpl
+++ b/test/constrained-generics/basic/set2/ref-intents-1.chpl
@@ -1,7 +1,7 @@
 
 interface IFC {
   proc req(ref arg: Self): void;
-  proc write(arg: Self): void;
+  proc write(arg: Self): bool;
 }
 
 int implements IFC;
```

File: modules/standard/ChapelIO.chpl (hunk 1 above)
Comment (12mohaned) [reply modules/standard/ChapelIO.chpl:840]: I believe the curl braces make the procedure more readable other than that I do not believe they do matters.

**Comment by vasslitvinov** [comment]
@12mohaned please remove a few files that you added perhaps unintentionally, like *.icloud files.

**Pull request closed without merging**